hasjob/models/filtered_view.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+from flask import url_for
+from sqlalchemy.dialects import postgresql
+from . import db, BaseScopedNameMixin, JobType, JobCategory, Tag, Domain, Board
+
+__all__ = ['FilteredView']
+
+
+filteredview_jobtype_table = db.Table('filteredview_jobtype_table', db.Model.metadata,
+    db.Column('filtered_view_id', None, db.ForeignKey('filtered_view.id'), primary_key=True),
+    db.Column('jobtype_id', None, db.ForeignKey('jobtype.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filteredview_jobcategory_table = db.Table('filteredview_jobcategory_table', db.Model.metadata,
+    db.Column('filtered_view_id', None, db.ForeignKey('filtered_view.id'), primary_key=True),
+    db.Column('jobcategory_id', None, db.ForeignKey('jobcategory.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filteredview_tag_table = db.Table('filteredview_tag_table', db.Model.metadata,
+    db.Column('filtered_view_id', None, db.ForeignKey('filtered_view.id'), primary_key=True),
+    db.Column('tag_id', None, db.ForeignKey('tag.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+filteredview_domain_table = db.Table('filteredview_domain_table', db.Model.metadata,
+    db.Column('filtered_view_id', None, db.ForeignKey('filtered_view.id'), primary_key=True),
+    db.Column('domain_id', None, db.ForeignKey('domain.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+class FilteredView(BaseScopedNameMixin, db.Model):
+    """
+    Store filters to display a filtered set of jobs scoped by a board on SEO friendly URLs
+
+    Eg: `https://hasjob.co/f/machine-learning-jobs-in-bangalore`
+    """
+
+    __tablename__ = 'filtered_view'
+
+    board_id = db.Column(None, db.ForeignKey('board.id'), primary_key=True, index=True)
+    board = db.relationship(Board)
+    parent = db.synonym('board')
+
+    #: Welcome text
+    description = db.Column(db.UnicodeText, nullable=False, default=u'')
+
+    #: Associated job types
+    types = db.relationship(JobType, secondary=filteredview_jobtype_table)
+    #: Associated job categories
+    categories = db.relationship(JobCategory, secondary=filteredview_jobcategory_table)
+    tags = db.relationship(Tag, secondary=filteredview_tag_table)
+    domains = db.relationship(Domain, secondary=filteredview_domain_table)
+    location_geonameids = db.Column(postgresql.ARRAY(db.Integer(), dimensions=1), nullable=True)
```

Comment:
Are we keeping the `ARRAY`? Can it be indexed for looking up filtered views given matching locations? It'll be a pain to migrate later if it's not suitable.

Reply:
`ARRAY` is cool, but add an index too. (The join tables are indexed thanks to the `primary_key` declarations).